hydra-node/src/Hydra/Chain/Direct/Handlers.hs

```diff
@@ -144,8 +145,14 @@ mkChain tracer queryTimeHandle wallet ctx LocalChainState{getLatest} submitTx =
           -- to bootstrap the init transaction. For now, we bear with it and
           -- keep the static keys in context.
           atomically (prepareTxToPost timeHandle wallet ctx chainState tx)
-            >>= finalizeTx wallet ctx chainState
+            >>= finalizeTx wallet ctx chainState mempty
         submitTx vtx
+    , draftTx = \utxo -> do
+        chainState <- atomically getLatest
+        case Hydra.Chain.Direct.State.chainState chainState of
+          Initial st ->
+            finalizeTx wallet ctx chainState utxo `traverse` commit ctx st utxo
```

Comment:
Maybe we should limit not allow to commit from the internal wallet owned UTxO, or maybe this is exactly what we want to replace the "internal" commit by this "external" commit.